File: Tool/PreUtil/src/output/c/medium/ChangeMinimumCharactersToSatisfyOneOfThreeConditions.c
/**https://leetcode.com/problems/change-minimum-characters-to-satisfy-one-of-three-conditions/ */
//You are given two strings a and b that consist of lowercase letters. In one operation, you can change any character in a or b to any lowercase letter.
//Your goal is to satisfy one of the following three conditions:
//Every letter in a is strictly less than every letter in b in the alphabet.
//Every letter in b is strictly less than every letter in a in the alphabet.
//Both a and b consist of only one distinct letter.
//Return the minimum number of operations needed to achieve your goal.


int minCharacters(char * a, char * b){
    int aLen = strlen(a);
    int bLen = strlen(b);
    int* aCount = (int*)calloc(26, sizeof(int));
    int* bCount = (int*)calloc(26, sizeof(int));
    int* aCount2 = (int*)calloc(26, sizeof(int));
    int* bCount2 = (int*)calloc(26, sizeof(int));
    int* aCount3 = (int*)calloc(26, sizeof(int));
    int* bCount3 = (int*)calloc(26, sizeof(int));
    int* aCount4 = (int*)calloc(26, sizeof(int));
    int* bCount4 = (int*)calloc(26, sizeof(int));
    int* aCount5 = (int*)calloc(26, sizeof(int));
    int* bCount5 = (int*)calloc(26, sizeof(int));
    int* aCount6 = (int*)calloc(26, sizeof(int));
    int* bCount6 = (int*)calloc(26, sizeof(int));
    int* aCount7 = (int*)calloc(26, sizeof(int));
    int* bCount7 = (int*)calloc(26, sizeof(int));
    int* aCount8 = (int*)calloc(26, sizeof(int));
    int* bCount8 = (int*)calloc(26, sizeof(int));
    int* aCount9 = (int*)calloc(26, sizeof(int));
    int* bCount9 = (int*)calloc(26, sizeof(int));
    int* aCount10 = (int*)calloc(26, sizeof(int));
    int* bCount10 = (int*)calloc(26, sizeof(int));
    int* aCount11 = (int*)calloc(26, sizeof(int));
    int* bCount11 = (int*)calloc(26, sizeof(int));
    int* aCount12 = (int*)calloc(26, sizeof(int));
    int* bCount12 = (int*)calloc(26, sizeof(int));
    int* aCount13 = (int*)calloc(26, sizeof(int));
    int* bCount13 = (int*)calloc(26, sizeof(int));
    int* aCount14 = (int*)calloc(26, sizeof(int));
    int* bCount14 = (int*)calloc(26, sizeof(int));
    


}

    

File: Tool/PreUtil/src/output/javascript/medium/ChangeMinimumCharactersToSatisfyOneOfThreeConditions.js
/**https://leetcode.com/problems/change-minimum-characters-to-satisfy-one-of-three-conditions/ */
//You are given two strings a and b that consist of lowercase letters. In one operation, you can change any character in a or b to any lowercase letter.
//Your goal is to satisfy one of the following three conditions:
//Every letter in a is strictly less than every letter in b in the alphabet.
//Every letter in b is strictly less than every letter in a in the alphabet.
//Both a and b consist of only one distinct letter.
//Return the minimum number of operations needed to achieve your goal.


/**
 * @param {string} a
 * @param {string} b
 * @return {number}
 */
var minCharacters = function(a, b) {
    let map1 = new Map();
    let map2 = new Map();
    let res = Infinity;
    for(let i = 0; i < a.length; i++){
        if(map1.has(a[i])){
            map1.set(a[i], map1.get(a[i]) + 1);
        }else{
            map1.set(a[i], 1);
        }
    }
    for(let i = 0; i < b.length; i++){
        if(map2.has(b[i])){
            map2.set(b[i], map2.get(b[i]) + 1);
        }else{
            map2.set(b[i], 1);
        }
    }
    let count1 = 0;
    let count2 = 0;
    for(let i = 0; i < 26; i++){
        let c = String.fromCharCode(97 + i);
        if(map1.has(c)){
            count1 += map1.get(c);
        }
        if(map2.has(c)){
            count2 += map2.get(c);
        }
        res = Math.min(res, a.length + b.length - count1 - count2);
    }
    count1 = 0;
    count2 = 0;
    for(let i = 0; i < 25; i++){
        let c = String.fromCharCode(97 + i);
        if(map1.has(c)){
            count1 += map1.get(c);
        }
        if(map2.has(c)){
            count2 += map2.get(c);
        }
        res = Math.min(res, a.length - count1 + count2);
        res = Math.min(res, b.length - count2 + count1);
    }
    return res;
    
};

    